### PR TITLE
try to get userID from URL if not present in query params

### PR DESCRIPTION
**Pull request by zkokelj**
### Why this change is needed

We need to add hardhat support for using Obscuro Gateway: https://github.com/obscuronet/obscuro-internal/issues/2171

Currently hardhat is not using queryParameters and this prevents users from using it via Obscuro Gateway.
We decided do provide a way by adding `/<userID>` at the end of the request URL.

This is a temporary solution, because HTTPS does not encrypt URL (but encrypts query parameters).

During testing I found and fixed another small mistake on frontend (we need to sign address in lowercase)

### What changes were made as part of this PR

Please provide a high level list of the changes made

### PR checks pre-merging

Please indicate below by ticking the checkbox that you have read and performed the required
[PR checks](https://github.com/obscuronet/obscuro-internal/blob/main/dev-ops-docs/dev-pr-checks.md)

- [ ] PR checks reviewed and performed

### CODE REVIEW

File: tools/walletextension/api/utils.go [hunk 1]
```diff
@@ -46,3 +48,33 @@ func getQueryParameter(params map[string]string, selectedParameter string) (stri
...
+	// try getting userID from query parameters and return it if successful
+	userID, err := getQueryParameter(conn.ReadRequestParams(), common.UserQueryParameter)
+	if err == nil {
+		if len(userID) != common.MessageUserIDLen {
+			return "", fmt.Errorf(fmt.Sprintf("wrong length of userID from URL. Got: %d, Expected: %d", len(userID), common.MessageUserIDLen))
+		}
+		return userID, err
+	}
+
+	// Alternatively, try to get it from URL path
```
Comment (tudor-malene) [inline tools/walletextension/api/utils.go:62]: pls add a comment here that this is a temporary hack to work around a hard hat bug

**Review by tudor-malene — COMMENTED** [review]
lgtm.
Pls add comment

File: tools/walletextension/api/utils.go (hunk 1 above)
Comment (zkokelj) [reply tools/walletextension/api/utils.go:62]: updated

**Review by tudor-malene — APPROVED** [review]
lgtm

**Pull request merged**